docs/core-concepts/1204-secrets.md

```diff
@@ -5,15 +5,23 @@ displayed_sidebar: '0.2'
 
 # How to use secrets
 
+## What are Secrets?
+
+Secrets support in Dagger allows you to utilize confidential information -- such as passwords, API keys, SSH keys, etc -- when running your Dagger Plans, _without_ exposing those secrets in plaintext logs, writing them into the filesystem of containers you're building, or inserting them into cache.
+
+Secrets are never merged into the CUE configuration. They are managed by the Dagger runtime, only  referenced as opaque identifiers, and can only be used by a special filesystem mount or as an environment variable designed to minimize leak risk. If you're interested in more details, these [Buildkit docs](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypesecret) describe them in more depth.
```

Comment:
Yeah, that's a good idea, @sipsma. Do you think it's worthwhile to link out to them? I was going back and forth on it.

Reply:
Actually now that you ask and I opened the link up, maybe it would be better not to. It doesn't provide a ton of information (the docs you wrote are better) and I think it could be confusing for users who aren't familiar with the difference between dagger+buildkit (e.g. they might get confused and think that secrets are something specific to dockerfiles).

Reply:
Thanks for the confirmation, @sipsma. I'll pull it out!